sdk/core/Azure.Core/tests/TestFramework/RetryTransport.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Azure.Core.Pipeline;
+
+namespace Azure.Core.Testing
+{
+    public class RetryTransport : HttpPipelineTransport
```

Comment:
This needs a better name. Something like `FailureSimulationTransport` but better :)

Reply:
`SimulateRetryTransport`?

Reply:
The transport doesn't simulate retries, it simulates failures thought. Retries just happen naturally.